log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/action/AbstractPathAction.java

```diff
@@ -103,7 +103,15 @@ public boolean execute(final FileVisitor<Path> visitor) throws IOException {
      * @return the base path (all lookups resolved)
      */
     public Path getBasePath() {
-        return Paths.get(subst.replace(getBasePathString()));
+        Path path = Paths.get(subst.replace(getBasePathString()));
+        if (Files.isSymbolicLink(path)) {
+            try {
+                path = Files.readSymbolicLink(path);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Printing the stack trace is a no-no. Either recover from the problem, propagate, or log to the status logger, I'm not sure which is best here.

Reply:
Also, should the replace call happen, before, after, or both?

Reply:
ok, I think we should throw out the exception.  
It's necessary to capture the exception , because this IOException has been caught outside.
The status logger can also be printed when handling exceptions externally

Reply:
I hava changed code，but  compile fail，do you know why? By the way，it compile normally  in my local.